docs/how_to_generate_umm_var_via_cmr_graphql.ipynb

```diff
@@ -0,0 +1,296 @@
+    "* `definition`\n",
+    "* `dimensions`\n",
+    "* `longName`\n",
+    "* `name`\n",
+    "* `standardName`\n",
+    "* `units`\n",
+    "* `metadataSpecification`\n",
+    "\n",
+    "For more information on each of these fields, please see the [UMM-Var JSON schema](https://git.earthdata.nasa.gov/projects/EMFD/repos/unified-metadata-model/browse/variable/v1.8.2/umm-var-json-schema.json)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "2a584730-66e1-43c4-90aa-2ebdc2dd02fb",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "graphql_query = '''\n",
+    "    query Collection($params: CollectionInput) {\n",
+    "      collection(params: $params) {\n",
+    "        generateVariableDrafts {\n",
+    "          count\n",
+    "          items {\n",
+    "            dataType\n",
+    "            definition\n",
+    "            dimensions\n",
+    "            longName\n",
+    "            name\n",
+    "            standardName\n",
+    "            units\n",
+    "            metadataSpecification\n",
+    "          }\n",
+    "        }\n",
+    "      }\n",
+    "    }\n",
+    "'''"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3f68f0fc-41d2-4bc2-bcb5-91d3f091f66f",
+   "metadata": {},
+   "source": [
+    "## Set GraphQL query parameters:\n",
+    "\n",
+    "The `Collection` query in CMR-GraphQL has a number of parameters that can be specified to identify matching collections. In this example, the collection concept ID is specified. The example collection used [a testing copy of a GPM IMERG precipitation product](https://cmr.uat.earthdata.nasa.gov/search/concepts/C1245618475-EEDTEST.html)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8696a4de-56cb-4524-8b04-f7b1e04f1d29",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "variables = {\n",
+    "    'params': {\n",
+    "        'conceptId': 'C1245618475-EEDTEST'\n",
+    "    }\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f164de17-39f1-4db3-b077-fe0e72b21c5e",
+   "metadata": {},
+   "source": [
+    "## Create full JSON payload:\n",
+    "\n",
+    "With the query and the variables defined, the JSON payload for the HTTP request to the CMR-GraphQL interface can be formed:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "6b35c3cb-43bb-4d5b-ba69-c52e8f1ff0f4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "payload = {\n",
+    "    'query': graphql_query,\n",
+    "    'variables': variables\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "62c1b142-8b21-4d74-b9ff-761b94483dc2",
+   "metadata": {},
+   "source": [
+    "## Configure HTTP request headers:\n",
+    "\n",
+    "The HTTP request made in this notebook requires two headers:\n",
+    "\n",
+    "* `Authorization` - this will include an [Earthdata Login](https://urs.earthdata.nasa.gov/) (EDL) Bearer token.\n",
```

Comment:
1. Yup. LaunchPad tokens are valid for auth, but they are a bit of a pain to set up. (I should probably include that link you have in your notebook for that)
2. Nope - I left the code cells mostly unchanged (except for PEP8 stuff, of course), but my preferred way would be to retrieve an token from EDL via `requests`, but I thought that would clutter up the notebook and distract the user from the meat of the documentation, because it's probably another 20 lines or so of code. I could implement that instead, or I could add some text to describe a couple of alternative ways to get the token. (Do you have a preference?)
3. This is probably a personal preference, but prefer to have this right by the place where the token is being retrieved. Just because someone getting to this point of the notebook will have both the cell that does something with a token and the documentation about that token on the screen at the same time without scrolling.

- [ ] Add link for LaunchPad token documentation.
- [ ] Clarify methods to get a token.

Reply:
As discussed in Slack, I added a helper function that retrieves an EDL token programmatically instead of using a text file. (I like this better, but hopefully the notebook is still fairly concise)

Reply:
Yes! I think it's a lot better and I will definitely be using that edl helper function!